File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.ts
function longestConsecutive(nums: number[]): number {
  if (nums.length === 0) return 0;
  let sort = [1];
  
  nums.sort((a, b) => a - b);
  nums.forEach((v, i, a) => {
    const diff = v - a[i - 1];
    if (diff === 0) sort;
    else if (diff === 1) sort[sort.length - 1]++;
    else sort.push(1);
  })
  return Math.max(...sort);
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> set;
        
        for (int num : nums) {
            set.insert(num);
        }
        
        int res = 0;
        
        for (int num: set) {
            if (!set.contains(num - 1)) {
                int cur = num;
                int streak = 1;
                
                while (set.contains(cur + 1)) {
                    cur++;
                    streak++;
                }
                
                res = max(res, streak);
            }
        }
        
        return res;
    }
};